js/lib/beautify.js

```diff
@@ -426,6 +427,11 @@
             if ((opt.preserve_newlines && current_token.wanted_newline) || force_linewrap) {
                 print_newline(false, true);
             } else if (opt.wrap_line_length) {
+                if (_newline_restricted_tokens.test(flags.last_text)) {
```

Comment:
You don't want to use just `last_text` here, you also need `last_type === 'TK_RESERVED'`.